deltacat/tests/compute/compact_partition_test_cases.py

```diff
@@ -0,0 +1,450 @@
+    "7-incremental-timestamp-pk-simple": IncrementalCompactionTestCase(
+        primary_keys_param={"pk_col_1"},
+        sort_keys_param=[
+            {
+                "key_name": "sk_col_1",
+            },
+        ],
+        partition_keys_param=[{"key_name": "region_id", "key_type": "int"}],
+        partition_values_param=["1"],
+        column_names_param=["pk_col_1", "sk_col_1"],
+        input_deltas_arrow_arrays_param=[
+            pa.array(offer_iso8601_timestamp_list(10, "minutes")),
+            pa.array([str(i) for i in range(10)]),
+        ],
+        input_deltas_delta_type=DeltaType.UPSERT,
+        expected_terminal_compact_partition_result=pa.Table.from_arrays(
+            [
+                pa.array(offer_iso8601_timestamp_list(10, "minutes")),
+                pa.array([str(i) for i in range(10)]),
+            ],
+            names=["pk_col_1", "sk_col_1"],
+        ),
+        validation_callback_func=None,
+        validation_callback_func_kwargs=None,
+        create_placement_group_param=True,
+        records_per_compacted_file_param=DEFAULT_MAX_RECORDS_PER_FILE,
+        hash_bucket_count_param=None,
+        create_table_strategy=create_src_w_deltas_destination_strategy,
+    ),
+    "8-incremental-decimal-timestamp-pk-multi": IncrementalCompactionTestCase(
+        primary_keys_param={"pk_col_1", "pk_col_2"},
+        sort_keys_param=[
+            {
+                "key_name": "sk_col_1",
+            },
+        ],
+        partition_keys_param=[{"key_name": "region_id", "key_type": "int"}],
+        partition_values_param=["1"],
+        column_names_param=["pk_col_1", "pk_col_2", "sk_col_1"],
+        input_deltas_arrow_arrays_param=[
+            pa.array([i / 10 for i in range(0, 20)]),
+            pa.array(offer_iso8601_timestamp_list(20, "minutes")),
+            pa.array([0.1] * 4 + [0.2] * 4 + [0.3] * 4 + [0.4] * 4 + [0.5] * 4),
+        ],
+        input_deltas_delta_type=DeltaType.UPSERT,
+        expected_terminal_compact_partition_result=pa.Table.from_arrays(
+            [
+                pa.array([i / 10 for i in range(0, 20)]),
+                pa.array(offer_iso8601_timestamp_list(20, "minutes")),
+                pa.array([0.1] * 4 + [0.2] * 4 + [0.3] * 4 + [0.4] * 4 + [0.5] * 4),
+            ],
+            names=["pk_col_1", "pk_col_2", "sk_col_1"],
+        ),
+        validation_callback_func=None,
+        validation_callback_func_kwargs=None,
+        create_placement_group_param=True,
+        records_per_compacted_file_param=DEFAULT_MAX_RECORDS_PER_FILE,
+        hash_bucket_count_param=None,
+        create_table_strategy=create_src_w_deltas_destination_strategy,
+    ),
+    "9-incremental-decimal-pk-multi-dup": IncrementalCompactionTestCase(
+        primary_keys_param={"pk_col_1"},
+        sort_keys_param=[
+            {
+                "key_name": "sk_col_1",
+            },
+        ],
+        partition_keys_param=[{"key_name": "region_id", "key_type": "int"}],
```

Comment:
Should we also include partition-less tables?

Reply:
Done

Reply:
Added

```
"10-incremental-decimal-pk-partitionless": IncrementalCompactionTestCase(
        primary_keys={"pk_col_1"},
        sort_keys=[SortKey.of(key_name="sk_col_1")],
        partition_keys_param=ZERO_VALUED_PARTITION_KEYS_PARAM,
        partition_values_param=ZERO_VALUED_PARTITION_VALUES_PARAM,
        column_names_param=["pk_col_1", "sk_col_1"],
        input_deltas_arrow_arrays_param=[
            pa.array([0.1] * 4 + [0.2] * 4 + [0.3] * 4 + [0.4] * 4 + [0.5] * 4),
            pa.array([i for i in range(20)]),
        ],
        input_deltas_delta_type=DeltaType.UPSERT,
        expected_terminal_compact_partition_result=pa.Table.from_arrays(
            [
                pa.array([0.1, 0.2, 0.3, 0.4, 0.5]),
                pa.array([3, 7, 11, 15, 19]),
            ],
            names=["pk_col_1", "sk_col_1"],
        ),
        validation_callback_func=None,
        validation_callback_func_kwargs=None,
        create_placement_group_param=False,
        records_per_compacted_file_param=DEFAULT_MAX_RECORDS_PER_FILE,
        hash_bucket_count_param=None,
        create_table_strategy=create_src_w_deltas_destination_strategy,
    ),
```